### DISCUSSION TITLE
Downgrade Development to Stable Version?

### DISCUSSION THREAD

**Discussion by 0-dot** [LMS-Community/slimserver#888] [Q&A]
I have installed the development version as a test and want to go back to the stable version. The upgrade had worked fine; since I use multiple players and have numerous plugins configured, I don't want to lose the existing configuration when downgrading. 

Which folders or files do I need to backup, and which to delete, for a downgrade to work. Or maybe there is another way back to stable?

lmscommunity/logitechmediaserver:dev (v8.4.0)
-->
lmscommunity/logitechmediaserver:stable (v8.3.2)

Thanks for the support.

**Comment by michaelherger** [comment]
You should be able to simply install 8.3.x on top of 8.4. 

I guess you're using the Docker image? In that case simply make sure you map the same folders to the container

**Reply by 0-dot** [reply]
Thank you for the quick reply. Yes I am using Docker on a Synology. 

I have already tried a direct downgrade, replacing version 8.4.0 (Dev) with 8.3.2 (Stable). The port assignments as well as the folder assignments and contents remained unchanged - I only replaced the Docker container.

The following behavior can be observed after the downgrade:
After restarting the container, the display on the players and the web interface started properly. Even navigating in the music folders, radios, favorites, playlists, etc. is possible. As soon as I want to play a song, in the case of the radio, the display gets stuck for about 30 seconds with the message "Titelinfos werden geladen...", in the case of the music folders, the songs are loaded but nothing plays, even the progress bar does not move. It feels like something is stuck in any case immediately after the command to play the music.

After replacing the container with the dev version, everything works as before.

**Reply by michaelherger** [reply]
Do you have two containers sharing the same configuration, but only running one at any time? I think DSM's Docker management wouldn't allow to do this, would it? Please double check ports. This sounds as if port 3483 wasn't set up correctly. Maybe DSM re-assigned alternative ports?

**Reply by 0-dot** [reply]
Thanks again for looking into the problem further. You are correct, it probably must have something to do with the network connection. I tried running the network test in the player after the downgrade and had no transmission (the bars stayed blank). On the other hand, the network diagnostics (probably just a ping to the server?) was positive, this test was reported as passed and using the repair option for the network did not bring any change either.

Regarding the guess about Docker container handling: I always replace the previously stopped container-image with the other version, all referenced folders remain the same after the version change. So at the time of a new test, there is always only one container with the identical predefined settings in the configuration files, which remain unchanged in the same folder structure.

Could it be that possibly a connection state was cached that the two versions do not handle identically?

**Reply by michaelherger** [reply]
Did you check the ports being exposed by the container when the connection failed? And is there anything in the server.log file?

**Reply by 0-dot** [reply]
The ports are assigned correctly and the server log shows a client connection error that I can't fully interpret:

Slim::Web::Cometd::handler (431) errorNeedsClient: 172.20.0.1, 00:04:20:2b:0c:92, status, -, 10, menu:menu, useContextMenu:1, subscribe:600

The IP is the gateway address that Docker assigns to the network bridge. The IP address itself is 172.17.0.6. Kinda weird that we don't use the IP address itself to connect?

**Reply by mherger** [reply]
Maybe you want to join the community at https://forums.slimdevices.com. There are many more DSM/Docker/LMS users who might know better.

**Reply by 0-dot** [reply]
That seems like a good suggestion to me; thanks for all your time looking into this question.

Have a great day and thank you for your commitment to keep the slimserver running on Docker.
